`zerocopy` 0.8 is missing `Hash` impls

When updating to `zerocopy` 0.8.4 for my project, it broke everything because `LE`, `BE`, etc do not implement `Hash`. The integer types like `U32<O>` have an impl for when `O: Hash`, but because `LE` and `BE` do not impl hash, it is unusable for me. I looked around but surprisingly it seems like no one else has this issue. There might be other things missing `Hash`.

For example,
```rust
use zerocopy::{FromBytes, Immutable, IntoBytes, KnownLayout, Unaligned, LE, U32};

#[derive(
    Clone, Copy, PartialEq, Eq, FromBytes, Immutable, IntoBytes, KnownLayout, Unaligned, Hash,
)]
#[repr(C)]
pub struct Test {
    pub test: U32<LE>,
}
```
produces
```
error[E0277]: the trait bound `zerocopy::LittleEndian: Hash` is not satisfied
 --> src/main.rs:8:5
  |
4 |     Clone, Copy, PartialEq, Eq, FromBytes, Immutable, IntoBytes, KnownLayout, Unaligned, Hash,
  |                                                                                          ---- in this derive macro expansion
...
8 |     pub test: U32<LE>,
  |     ^^^^^^^^^^^^^^^^^ the trait `Hash` is not implemented for `zerocopy::LittleEndian`, which is required by `U32<zerocopy::LittleEndian>: Hash`
  |
  = help: the trait `Hash` is implemented for `U32<O>`
  = note: required for `U32<zerocopy::LittleEndian>` to implement `Hash`
  = note: this error originates in the derive macro `Hash` (in Nightly builds, run with -Z macro-backtrace for more info)
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@
 [package]
 edition = "2021"
 name = "zerocopy"
-version = "0.8.4"
+version = "0.8.5"
 authors = ["Joshua Liebow-Feeser <joshlf@google.com>"]
 description = "Zerocopy makes zero-cost memory manipulation effortless. We write \"unsafe\" so you don't have to."
 categories = ["embedded", "encoding", "no-std::no-alloc", "parsing", "rust-patterns"]
@@ -77,13 +77,13 @@ std = ["alloc"]
 __internal_use_only_features_that_work_on_stable = ["alloc", "derive", "simd", "std"]
 
 [dependencies]
-zerocopy-derive = { version = "=0.8.4", path = "zerocopy-derive", optional = true }
+zerocopy-derive = { version = "=0.8.5", path = "zerocopy-derive", optional = true }
 
 # The "associated proc macro pattern" ensures that the versions of zerocopy and
 # zerocopy-derive remain equal, even if the 'derive' feature isn't used.
 # See: https://github.com/matklad/macro-dep-test
 [target.'cfg(any())'.dependencies]
-zerocopy-derive = { version = "=0.8.4", path = "zerocopy-derive" }
+zerocopy-derive = { version = "=0.8.5", path = "zerocopy-derive" }
 
 [dev-dependencies]
 itertools = "0.11"
@@ -97,6 +97,6 @@ testutil = { path = "testutil" }
 # CI test failures.
 trybuild = { version = "=1.0.89", features = ["diff"] }
 # In tests, unlike in production, zerocopy-derive is not optional
-zerocopy-derive = { version = "=0.8.4", path = "zerocopy-derive" }
+zerocopy-derive = { version = "=0.8.5", path = "zerocopy-derive" }
 # TODO(#381) Remove this dependency once we have our own layout gadgets.
 elain = "0.3.0"

--- a/src/byteorder.rs
+++ b/src/byteorder.rs
@@ -63,6 +63,7 @@
 use core::{
     convert::{TryFrom, TryInto},
     fmt::{Binary, Debug, LowerHex, Octal, UpperHex},
+    hash::Hash,
     num::TryFromIntError,
 };
 
@@ -81,7 +82,7 @@ use super::*;
 ///
 /// [`U32<BigEndian>`]: U32
 pub trait ByteOrder:
-    Copy + Clone + Debug + Display + Eq + PartialEq + Ord + PartialOrd + private::Sealed
+    Copy + Clone + Debug + Display + Eq + PartialEq + Ord + PartialOrd + Hash + private::Sealed
 {
     #[doc(hidden)]
     const ORDER: Order;
@@ -104,7 +105,7 @@ pub enum Order {
 /// Big-endian byte order.
 ///
 /// See [`ByteOrder`] for more details.
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
 pub enum BigEndian {}
 
 impl ByteOrder for BigEndian {
@@ -121,7 +122,7 @@ impl Display for BigEndian {
 /// Little-endian byte order.
 ///
 /// See [`ByteOrder`] for more details.
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
 pub enum LittleEndian {}
 
 impl ByteOrder for LittleEndian {
@@ -1070,7 +1071,7 @@ mod tests {
     }
 
     trait ByteOrderType:
-        FromBytes + IntoBytes + Unaligned + Copy + Eq + Debug + From<Self::Native>
+        FromBytes + IntoBytes + Unaligned + Copy + Eq + Debug + Hash + From<Self::Native>
     {
         type Native: Native;
         type ByteArray: ByteArray;

--- a/zerocopy-derive/Cargo.toml
+++ b/zerocopy-derive/Cargo.toml
@@ -9,7 +9,7 @@
 [package]
 edition = "2021"
 name = "zerocopy-derive"
-version = "0.8.4"
+version = "0.8.5"
 authors = ["Joshua Liebow-Feeser <joshlf@google.com>"]
 description = "Custom derive for traits from the zerocopy crate"
 license = "BSD-2-Clause OR Apache-2.0 OR MIT"

